canister/src/unstable_blocks.rs

```diff
@@ -170,31 +178,32 @@ pub fn get_chain_with_tip<'a, 'b>(
 
 // Returns the index of the `anchor`'s stable child if it exists.
 fn get_stable_child(blocks: &UnstableBlocks) -> Option<usize> {
-    // Compute the depth of all the children.
-    let mut depths: Vec<_> = blocks
+    // Compute the normalized weight of all the children.
+    let network = blocks.network;
+    let mut weights: Vec<_> = blocks
         .tree
         .children
         .iter()
         .enumerate()
-        .map(|(idx, child)| (blocktree::depth(child), idx))
+        .map(|(idx, child)| (blocktree::get_normalized_weight(child, network), idx))
```

Comment:
This is incorrect. What this code is doing is summing the difficulties of each child, then normalizing those difficulties with the root of each child. What we want is to normalize with the `blocks.tree.root`. The fix here would be to call `get_max_weight` here, then in a later step in this method divide the difficulty of each one by `blocks.tree.root`.

I'd also add a test case to ensure that we are normalizing with `blocks.tree.root`.

Reply:
Maybe in that case it is better to avoid division and just multiply stability treshlod with `blocks.tree.root.difficulty()`, because it will give the same result.

Reply:
Yes, multiplying the threshold with the root difficulty would work as well (without the use of division).